### PR TITLE
Add socket buffering

### PR DESCRIPTION
**Pull request by tomato42**
reduce number of system calls in the handshake phase by buffering the messages until we start waiting for the other side to respond

---
This change is [Reviewable](https://reviewable.io/reviews/tomato42/tlslite-ng/100)

### CODE REVIEW

**Comment by tomato42** [comment]
@The-Mule: r?

**Comment by tomato42** [comment]
@joshbressers: r?

File: tlslite/bufferedsocket.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+    def send(self, data):
+        """Send data to the socket"""
+        if self.buffer_writes:
+            self._write_queue.append(data)
+            return len(data)
+        return self.socket.send(data)
+
+    def sendall(self, data):
+        """Sand data to the socket"""
```
Comment (joshbressers) [inline tlslite/bufferedsocket.py]: s/Sand/Send/

**Comment by joshbressers** [comment]
Once you fix the comment typo above, this gets an r+ from me.

**Pull request merged**